### PR TITLE
(kubernetes) support deployment API

### PR DESCRIPTION
**Pull request by lwander**
@duftler PTAL - this only addresses deploy, delete & rollback haven't been completed yet.

### CODE REVIEW

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiConverter.groovy [hunk 1]
```diff
@@ -756,4 +755,108 @@ class KubernetesApiConverter {
...
+      .endSelector()
+
+    def targetSize = description.targetSize ?: description.capacity?.desired
+
+    serverGroupBuilder = serverGroupBuilder.withReplicas(targetSize)
+      .withNewTemplate()
+      .withNewMetadata()
+      .addToLabels(labels + matchLabels)
+
+    for (def loadBalancer : description.loadBalancers) {
```
Comment (duftler) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiConverter.groovy]: Could use `description.loadBalancers.each { }` for these kinds of loops.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/description/servergroup/DeployKubernetesAtomicOperationDescription.groovy [hunk 2]
```diff
@@ -96,6 +97,37 @@ class KubernetesContainerDescription {
 
 @AutoClone
 @Canonical
+class KubernetesDeployment {
+  boolean enabled
+  KubernetesStrategy deploymentStrategy
+  int minReadySeconds
```
Comment (duftler) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/description/servergroup/DeployKubernetesAtomicOperationDescription.groovy]: Is there a reason to alternate between the primitive and wrapper types here? If so, please leave a comment or two for future readers.

Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/description/servergroup/DeployKubernetesAtomicOperationDescription.groovy]: Added some comments - the nullable values are optional

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiConverter.groovy (hunk 1 above)
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiConverter.groovy]: This is one of those groovy bugs where the closure erases some class values

**Pull request merged**